editor/property_selector.cpp

```diff
@@ -575,7 +575,29 @@ void PropertySelector::set_type_filter(const Vector<Variant::Type> &p_type_filte
 	type_filter = p_type_filter;
 }
 
+//Variant::Type isn't exposed, so need a version that takes int
```

Comment:
Indeed: [here](https://docs.godotengine.org/en/latest/classes/class_%40globalscope.html#enum-globalscope-variant-type)

Reply:
Thanks for the review!

Ah, I see, that makes sense!

Nevertheless, I am getting an error from `ClassDB::bind_method` when I try to bind `set_type_filter` instead of `set_type_filter_exposable`

```
In template: implicit instantiation of undefined template 'GetTypeInfo<const Vector<Variant::Type> &>'
```

Is it maybe because `Variant::Type` isn't itself a `Variant` so it can't be implicitly converted from `Vector<Variant::Type>` to `TypedArray<Variant::Type>`?

Reply:
You need to use `TypedArray`, `Vector` will only work if it's a packed case I think

Reply:
I tried that first actually, and it also gave an error. I tried it again just now, and this is the error I get:

```
In template: no member named 'get_class_static' in 'Variant::Type'
```

Thanks for helping me out with this :)

Reply:
Then I guess you might have to restrict yourself to `Vector<int>` here

Reply:
Okay. Any ideas for a better comment to explain the choice? Maybe `Need a version that takes int because TypedArray<Variant::Type> is not valid.`

Reply:
`TypedArray<int> because TypedArray<Variant::Type> is not supported.`

Reply:
I didn't realize `Variant::Type` couldn't be used with `TypedArray<T>`. Would it be possible to make it work? If so, it'd be nice to do it before shipping this API; otherwise, we won't be able to change it later to avoid breaking compat.

Reply:
My guess is that it's because `Variant::Type` isn't a variant, and TypedArray can only hold Variants.  I could be totally wrong here though. Who would be the best person to ask about it?

Reply:
Possible that it's caused by a lack of enum conversion, will look into it